src/h5web/dataset-visualizer/VisDisplay.tsx

```diff
@@ -15,13 +15,10 @@ interface Props {
 
 function VisDisplay(props: Props): JSX.Element {
   const { activeVis, dataset } = props;
-  const { Component: VisComponent } = VIS_DEFS[activeVis];
-
   const datasetDims = isSimpleShape(dataset.shape) ? dataset.shape.dims : [];
-  const [mapperState, setMapperState] = useState<DimensionMapping | undefined>(
-    undefined
-  );
-  const [prevVis, setPrevVis] = useState(activeVis);
+
+  const [prevVis, setPrevVis] = useState<Vis>();
```

Comment:
By initialising `prevState` to nothing (i.e. `undefined`), we go through the derived state `if` block below on first render, which in turn initialises `mapperState` appropriately.  Before, we weren't going through the `if` block, so the `mapperState` was initialised to `undefined`.

In `DatasetVisualizer` above, I had worked around the issue by duplicating the logic in the initial state. I think going through the derived state `if` block is a cleaner solution (especially since rendering of the children is canceled).